src/CodeGen_D3D12Compute_Dev.cpp

```diff
@@ -455,7 +455,7 @@ void CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_C::visit(const Store *op) {
     Type value_type = op->value.type();
 
     // __shared[x] is always uint(32): must reinterpret/pack bits...
-    if (op->name == "__shared") {
```

Comment:
We really want to eliminate all string matching from the compiler. It's a constant source of bugs.

I suggest giving this class a `Scope<> shared_allocations` member, and then checking shared_allocations.contains(op->name) here.

Reply:
> We really want to eliminate all string matching from the compiler. 

See also: https://wiki.c2.com/?StringlyTyped